src/index.js

```diff
@@ -15,29 +13,77 @@ const displayItem = () => {
     const attr = document.createAttribute('data-id');
     attr.value = todo.index;
     div.setAttributeNode(attr);
-
+    //   creating div for checkbox and input
     const divElement = document.createElement('div');
     divElement.className = 'checkText';
+    // creating checkbox
     const inputElement = document.createElement('input');
     inputElement.type = 'checkbox';
     inputElement.id = 'myCheckbox';
+    //   creating text input
     const inputTextElement = document.createElement('input');
     inputTextElement.classList.add('toDoText');
     inputTextElement.type = 'text';
     inputTextElement.value = todo.description;
+    // event listener for edit
+    inputTextElement.addEventListener('blur', () => {
+      const newInputValue = inputTextElement.value;
+      todo.description = newInputValue;
+      localStorage.setItem('arrayOfTasks', JSON.stringify());
+      localStorage.setItem('arrayOfTasks', JSON.stringify(arrayOfTasks));
+    });
+    //   appending checkbox and text input to the div
     divElement.appendChild(inputElement);
     divElement.appendChild(inputTextElement);
-    const buttonElement = document.createElement('button');
-    buttonElement.type = 'button';
-    buttonElement.className = 'btn';
+    //   creating icon
     const iElement = document.createElement('i');
-    iElement.className = 'fa-solid fa-ellipsis-vertical';
-    buttonElement.appendChild(iElement);
+    iElement.classList.add('fa-solid');
+    iElement.classList.add('fa-ellipsis-vertical');
+    // event listener for three dot
+    iElement.addEventListener('click', () => {
+      iElement.classList.toggle('fa-solid');
+      iElement.classList.toggle('fa-ellipsis-vertical');
+      iElement.classList.toggle('fa-trash-can');
+      iElement.classList.toggle('fa-regular');
+      //   event listener for delete item
+      iElement.addEventListener('click', () => {
+        const item = iElement.parentElement;
+        const itemId = item.getAttribute('data-id');
+        arrayOfTasks = arrayOfTasks.filter((task) => task.index !== parseInt(itemId, 10));
+        toDoList.removeChild(item);
+        arrayOfTasks.forEach((obj, index) => {
+          if (obj.index > parseInt(itemId, 10)) {
+            obj.index = index + 1;
+          }
+        });
+        localStorage.setItem('arrayOfTasks', JSON.stringify(arrayOfTasks));
+      });
+    });
+
+    //   buttonElement.appendChild(iElement);
```

Comment:
- [ ] You have made most of the project requirements 👍 **Good job so far 👏 👏** However, It would be better to add **a new JavaScript file** for the new functionality just to keep your code organized. 😎😎

    _If you have a lot of JavaScript code in one file, it can become difficult to manage and maintain. By splitting your code into separate files, you can make it easier to find and understand the code that you need. 👍_